### PR TITLE
Axios removal

### PR DESCRIPTION
**Pull request by nivvyart**
pull request for review

### CODE REVIEW

File: src/components/stats/SoloCommits.js [hunk 1]
```diff
@@ -16,80 +16,81 @@ class SoloCommits extends Component {
...
-      });
-      Promise.all(requests).then(() => {
-        // TODO: call findHighest
-        this.findHigest();
-      });
-    });
-  }
-
-  findHigest() {
+  findHigest(data) {
```
Comment (jamesla) [inline src/components/stats/SoloCommits.js]:
```
findhighest(data)
```

File: src/components/stats/SoloCommits.js [hunk 2]
```diff
@@ -16,80 +16,81 @@ class SoloCommits extends Component {
...
+    //setting state here crashes react, you cant set the state multiple times from within the render
+    // this.setState({ winner: winner });
+    // this.setState({ highScore: testing[winner] });
+
+    //i need an object here that has more that winner and commit total so i can use that to build the
+    // card, rather than recalling GQL like in the axios example
   }
 
   render() {
-    if (!this.state.winner) return <p>loading...</p>;
```
Comment (jamesla) [inline src/components/stats/SoloCommits.js]: should use deconstructed props here 

```
const { winner, username, etc } = this.state;
```

File: src/components/stats/SoloCommits.js [hunk 3]
```diff
@@ -16,80 +16,81 @@ class SoloCommits extends Component {
...
-            user(login: "${this.state.winner}") {
-              avatarUrl
-              url
-
-              followers {
-                totalCount
-              }
-              starredRepositories {
-                totalCount
+            repository(owner: "${this.state.username}", name: "${
```
Comment (jamesla) [inline src/components/stats/SoloCommits.js]: no need to string interpolate this:
```
repository(owner: this.state.username, name: this.state.repository") ...
```

and with the prop assignment fix from above:

```
repository(owner: username, name: repository") ...
```

File: src/components/stats/SoloCommits.js [hunk 4]
```diff
@@ -16,80 +16,81 @@ class SoloCommits extends Component {
...
+          this.state.repository
+        }") {
+              defaultBranchRef {
+                target {
+                  ... on Commit {
+                    history(
+                      since: "${this.state.startDate}"
```
Comment (jamesla) [inline src/components/stats/SoloCommits.js]: remove string interpolation and use restructured props

File: src/components/stats/SoloCommits.js [hunk 5]
```diff
@@ -16,80 +16,81 @@ class SoloCommits extends Component {
...
+                      until: "${this.state.endDate}"
```
Comment (jamesla) [inline src/components/stats/SoloCommits.js]: remove string interpolation and use restructured props

File: src/components/stats/SoloCommits.js [hunk 6]
```diff
@@ -99,32 +100,13 @@ class SoloCommits extends Component {
           if (loading) return <p>Loading...</p>;
           if (error) return <p>Error </p>;
 
+          this.findHigest(data);
```
Comment (jamesla) [inline src/components/stats/SoloCommits.js]:
```
this.findHighest(data);
```

File: src/components/stats/SoloCommits.js [hunk 7]
```diff
@@ -99,32 +100,13 @@ class SoloCommits extends Component {
...
-                        Followers: {data.user.followers.totalCount}
-                      </li>
-                      <li className="list-inline-item">
-                        Starred Repositories:{" "}
-                        {data.user.starredRepositories.totalCount}
-                      </li>
-                    </ul>
-                  </div>
-                </div>
+                <h1>{this.state.winner}</h1>
```
Comment (jamesla) [inline src/components/stats/SoloCommits.js]: use destructured props

File: src/components/stats/SoloCommits.js [hunk 8]
```diff
@@ -99,32 +100,13 @@ class SoloCommits extends Component {
...
+                <h2>{this.state.highScore}</h2>
```
Comment (jamesla) [inline src/components/stats/SoloCommits.js]: use destructured props

File: src/components/stats/SoloCommits.jsx [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+          }
+        }
+      }`,
+    })
+      .then((result) => {
+        this.findHigest(result);
+      });
+  }
+
+  findHigest(data) {
```
Comment (jamesla) [inline src/components/stats/SoloCommits.jsx]: Spelling

File: src/components/Search.jsx [hunk 10]
```diff
@@ -40,17 +40,21 @@ class Search extends Component {
   }
 
   //need to add a default search value here if those fields are not added.
-  _selectRepo() {
-    this.props.history.push(
-      `/project/${this.state.username}/${this.state.repository}/${
+  _selectRepo(event) {
+    if (!this.state.username || !this.state.repository) {
```
Comment (jamesla) [inline src/components/Search.jsx]: Deconstruct props

File: src/components/Search.jsx [hunk 11]
```diff
@@ -40,17 +40,21 @@ class Search extends Component {
...
+    } else {
+      this.props.history.push(
```
Comment (jamesla) [inline src/components/Search.jsx]: Deconstruct this

File: src/components/Search.jsx [hunk 12]
```diff
@@ -40,17 +40,21 @@ class Search extends Component {
...
+        `/project/${this.state.username}/${this.state.repository}/${
```
Comment (jamesla) [inline src/components/Search.jsx]: Deconstruct this

File: src/components/Search.jsx [hunk 13]
```diff
@@ -40,17 +40,21 @@ class Search extends Component {
...
         this.state.startDate
```
Comment (jamesla) [inline src/components/Search.jsx]: Deconstruct this

File: src/components/Search.jsx [hunk 14]
```diff
@@ -40,17 +40,21 @@ class Search extends Component {
...
-      }/${this.state.endDate}`
+        }/${this.state.endDate}`,
```
Comment (jamesla) [inline src/components/Search.jsx]: Deconstruct this

File: src/components/Search.jsx [hunk 15]
```diff
@@ -40,17 +40,21 @@ class Search extends Component {
...
 
-      //this.props.match.params.usernmae/repo
-    );
+      );
+    }
   }
 
   render() {
+    //const { validated } = this.state;
```
Comment (jamesla) [inline src/components/Search.jsx]: Dont merge comments to master

File: src/components/stats/PullStats.jsx [hunk 16]
```diff
@@ -28,7 +28,6 @@ class PullStats extends Component {
         <Query
           query={gql`
                     {
-
                         repository(owner: "${this.state.username}", name: "${
             this.state.repository
```
Comment (jamesla) [inline src/components/stats/PullStats.jsx]: Deconstruct this Also run auto formatter on this file

File: src/components/stats/SoloCommits.jsx [hunk 17]
```diff
@@ -0,0 +1,137 @@
...
+  componentDidMount() {
+    const { username } = this.state;
+    const { repository } = this.state;
+    const { startDate } = this.state;
+    const { endDate } = this.state;
+
+    client.query({
+      query: gql`
+      {
+        repository(owner: "${username}", name: "${repository}") {
```
Comment (jamesla) [inline src/components/stats/SoloCommits.jsx:31]: Owner should already be a string so no need to string interpolate it (remove brackets and dollar sign around these variables)

File: src/components/stats/SoloCommits.jsx [hunk 18]
```diff
@@ -0,0 +1,137 @@
...
+                  }
+                }
+              }
+            }
+          }
+        }
+      }`,
+    })
+      .then((result) => {
+        this.findHigest(result);
```
Comment (jamesla) [inline src/components/stats/SoloCommits.jsx]: Spelling

File: src/components/stats/SoloCommits.jsx [hunk 19]
```diff
@@ -0,0 +1,137 @@
...
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }`,
+    })
```
Comment (jamesla) [inline src/components/stats/SoloCommits.jsx:50]: Small one but I’d prob just run your autoformatter on this file

File: src/components/stats/SoloCommits.jsx [hunk 20]
```diff
@@ -0,0 +1,137 @@
...
+
+  render() {
+    const { winner } = this.state;
+    const { highScore } = this.state;
+
+    return (
+      <Query
+        query={gql`
+          {
+            user(login: "${winner}") {
```
Comment (jamesla) [inline src/components/stats/SoloCommits.jsx:83]: String interpolation not needed (remove brackets dollar and quotes

File: src/components/stats/SoloRefactor.jsx [hunk 21]
```diff
@@ -0,0 +1,137 @@
...
+  componentDidMount() {
+    const { username } = this.state;
+    const { repository } = this.state;
+    const { startDate } = this.state;
+    const { endDate } = this.state;
+
+    client.query({
+      query: gql`
+      {
+        repository(owner: "${username}", name: "${repository}") {
```
Comment (jamesla) [inline src/components/stats/SoloRefactor.jsx:31]: String interpolation not needed (remove brackets dollar and quotes

File: src/components/stats/SoloRefactor.jsx [hunk 22]
```diff
@@ -0,0 +1,137 @@
...
+          defaultBranchRef {
+            target {
+              ... on Commit {
+                history(since: "${startDate}" until: "${endDate}") {
```
Comment (jamesla) [inline src/components/stats/SoloRefactor.jsx:35]: String interpolation not needed (remove brackets dollar and quotes

File: src/components/stats/SoloRefactor.jsx [hunk 23]
```diff
@@ -0,0 +1,137 @@
...
+                  }
+                }
+              }
+            }
+          }
+        }
+      }`,
+    })
+      .then((result) => {
+        this.findHigest(result);
```
Comment (jamesla) [inline src/components/stats/SoloRefactor.jsx]: Spelling

**Review by jamesla — CHANGES_REQUESTED** [review]
Code hygiene fixes

File: src/components/stats/SoloCommits.jsx (hunk 9 above)
Comment (nivvyart) [reply src/components/stats/SoloCommits.jsx]: updated

File: src/components/stats/SoloCommits.jsx (hunk 18 above)
Comment (nivvyart) [reply src/components/stats/SoloCommits.jsx]: updated

File: src/components/Search.jsx (hunk 10 above)
Comment (nivvyart) [reply src/components/Search.jsx]: updated

File: src/components/Search.jsx (hunk 11 above)
Comment (nivvyart) [reply src/components/Search.jsx]: deconstructed

File: src/components/Search.jsx (hunk 12 above)
Comment (nivvyart) [reply src/components/Search.jsx]: done

File: src/components/Search.jsx (hunk 13 above)
Comment (nivvyart) [reply src/components/Search.jsx]: done

File: src/components/Search.jsx (hunk 14 above)
Comment (nivvyart) [reply src/components/Search.jsx]: done

File: src/components/Search.jsx (hunk 15 above)
Comment (nivvyart) [reply src/components/Search.jsx]: removed

File: src/components/stats/PullStats.jsx (hunk 16 above)
Comment (nivvyart) [reply src/components/stats/PullStats.jsx]: need this for GQL, deconstructed at least

File: src/components/stats/SoloCommits.jsx (hunk 17 above)
Comment (nivvyart) [reply src/components/stats/SoloCommits.jsx:31]: i think GQL needs this, couldnt resolve.

File: src/components/stats/SoloCommits.jsx (hunk 19 above)
Comment (nivvyart) [reply src/components/stats/SoloCommits.jsx:50]: auto formatter wanted this... :)?

File: src/components/stats/SoloCommits.jsx (hunk 20 above)
Comment (nivvyart) [reply src/components/stats/SoloCommits.jsx:83]: GQL needs this

File: src/components/stats/SoloRefactor.jsx (hunk 21 above)
Comment (nivvyart) [reply src/components/stats/SoloRefactor.jsx:31]: gql needs this

File: src/components/stats/SoloRefactor.jsx (hunk 23 above)
Comment (nivvyart) [reply src/components/stats/SoloRefactor.jsx]: resolved

File: src/components/stats/SoloRefactor.jsx (hunk 22 above)
Comment (nivvyart) [reply src/components/stats/SoloRefactor.jsx:35]: will test more, i am pretty sure gql needs this specific formatting for its query

**Pull request merged**